dppy/vfx_sampling.py

```diff
@@ -110,57 +114,71 @@ def estimate_rls_with_embedding(eigvec, eigvals,
-                                              eval_L,
-                                              rls_oversample_dppvfx,
-                                              rls_oversample_bless,
-                                              rng,
-                                              desired_s,
-                                              H_bless)
-
-    precomputed_state.q = q_func(precomputed_state.s)
-
-    S, rej_count = _sampling_loop(X, eval_L, precomputed_state, rng)
+def vfx_sampling_precompute_constants(X,
+                                      eval_L,
+                                      rng,
+                                      desired_s=None,
+                                      rls_oversample_dppvfx=4.0,
+                                      rls_oversample_bless=4.0,
+                                      H_bless=None,
+                                      verbose=True):
+    """Pre-compute quantities necessary for the vfx rejection sampling loop, such as the inner Nystrom approximation,
+    and the RLS of all elements in L.
+
+        :param array_like X: dataset such that L = eval_L(X), out of which we are sampling objects according to a DPP
+        :param callable eval_L: likelihood function
+        :param RandomState rng: random source used for sampling
+        :param desired_s: desired expected sample size for the DPP. If None, use the natural DPP expected sample size.
+        The vfx sampling algorithm can adjust the expected sample size of the DPP by rescaling the L matrix with a
+        scalar alpha_star <= 1. Adjusting the expected sample size can be useful to control downstream complexity,
+        and it is necessary to improve the probability of drawing a sample with exactly k elements when
+        using vfx for k-DPP sampling. Currently only reducing the sample size is supported, and the sampler will
+        return an exception if the DPP sample has already a natural expected size smaller than desired_s.
+        :type:
+            float or None, default None
+        :param rls_oversample_dppvfx: Oversampling parameter used to construct dppvfx's internal Nystrom approximation.
+        The rls_oversample_dppvfx >= 1 parameter is used to increase the rank of the approximation by
+        a rls_oversample_dppvfx factor. This makes each rejection round slower and more memory intensive,
+        but reduces variance and the number of rounds of rejections, so the actual runtime might increase or decrease.
+        Empirically, a small factor rls_oversample_dppvfx = [2,10] seems to work. It is suggested to start with
+        a small number and increase if the algorithm fails to terminate.
+        :type rls_oversample_dppvfx:
+            float, default 4.0
+        :param rls_oversample_bless: Oversampling parameter used during bless's internal Nystrom approximation.
+        Note that this is a different Nystrom approximation than the one related to :ref:`rls_oversample_dppvfx`,
```

Comment:
Don't understand the `:ref:` here

Reply:
There are two different rls estimator used in different parts of bless and dpp-vfx, with slightly different semantics and guarantees. Depending on precomputed state available, one can be more efficient than the other, so we use both.

I added that comment to highlight the fact that that is not a redundant function, and we should keep both around, i.e. to avoid that one of the two version gets deleted in refactoring in the future without properly understanding that there is not two version of the same estimator.